cpp/include/cugraph/prims/transform_reduce_incoming_outgoing_e_of_v.cuh

```diff
@@ -985,7 +982,7 @@ template <typename GraphViewType,
           typename EdgeOp,
           typename T,
           typename VertexValueOutputIterator>
-void copy_v_transform_reduce_out_nbr(
+void transform_reduce_outgoing_e_of_v(
```

Comment:
`transform_reduce_outgoing_edges`?
The of_v part seems unnecessary to me.

Reply:
I would read transform_reduce_outgoing_e_of_v as 'transform reduce outgoing edges of vertices'.
The 'of vertices' doesn't really communicate that a per source reduction is being done. We should think of something that does make this clear.

We could name it exactly as transform_reduce_e to denote that every edge is being iterated on. The fact that this variant is doing per source reduction could be explained by differing function signature. This could be confusing to some users.

We could also use `transform_reduce_outgoing_edges_per_source` or '..._e_per_src'.

Reply:
Another point was that if we are talking in terms of edges, in my opinion we should use the term source/destination (src/dst) instead of vertex.
Perhaps I am overlooking some nuances when dealing with transpose (csc) graphs.

Reply:
Agreed, I like the `per_src` idea. I will make this change.

Reply:
Will `per_src` still make sense even when the graph is transposed?

Reply:
I assume so.

But `transform_reduce_outgoing_e_of_v` and `transform_reduce_e_per_src` may have slightly different mental models.

I initially hoped `outgoing_e_of_v` to be interpreted as "for every vertex, iterate over outgoing edges and reduce the edge functor outputs per vertex"; so this is more vertex centric.

"e_per_src" might be interpreted as iterate over every directed edge, apply a functor to each edge, than reduce functor outputs per source vertex ID; so this is more edge centric.

Both are pretty much doing the same thing, but nuances might be slightly different, and I feel like the latter is a bit clearer.

Reply:
Whether the graph is transposed or not affects efficiency of the primitive implementation. Per source reduction can be more efficient with a source-major format while per destination reduction can be more efficient with a destination-major format (by avoiding atomics, and some primitives allow only a source-major format... this is more to avoid implementing something that will not be used in the foreseeable future).

Reply:
My main concern about `per_src` nomenclature is about 0-(in|out)-degree vertices.

So these primitives are designed to serve `properties_of_v = some_initial_value + sum_over_incoming|outgoing_e (calculated per edge value)` (e.g. PageRank, Katz centrality, HITS, summing per vertex incoming/outgoing edge weights).

This doesn't align very well with the edge-centric mental model.